From af9d97fc3930b15669145c2096158158ae46b3fd Mon Sep 17 00:00:00 2001
From: Alex Elder <elder@inktank.com>
Date: Tue, 29 May 2012 21:47:38 -0500
Subject: [PATCH 502/938] libceph: kill bad_proto ceph connection op
References: fate#312983
Git-commit: 6384bb8b8e88a9c6bf2ae0d9517c2c0199177c34 (partial)
Patch-mainline: v3.6-rc1

No code sets a bad_proto method in its ceph connection operations
vector, so just get rid of it.

Signed-off-by: Alex Elder <elder@inktank.com>
Reviewed-by: Yehuda Sadeh <yehuda@inktank.com>
Acked-by: Danny Al-Gaaf <dalgaaf@suse.de>

---
 net/ceph/messenger.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/net/ceph/messenger.c b/net/ceph/messenger.c
index ad51b62..59047f3 100644
--- a/net/ceph/messenger.c
+++ b/net/ceph/messenger.c
@@ -1355,11 +1355,6 @@ static void fail_protocol(struct ceph_connection *con)
 {
 	reset_connection(con);
 	set_bit(CLOSED, &con->state);  /* in case there's queued work */
-
-	mutex_unlock(&con->mutex);
-	if (con->ops->bad_proto)
-		con->ops->bad_proto(con);
-	mutex_lock(&con->mutex);
 }
 
 static int process_connect(struct ceph_connection *con)
-- 
1.8.3

